### ISSUE TITLE
[gdal:x64-windows] broken due to #5839 

### ISSUE THREAD

**Issue by Neumann-A** [Microsoft/vcpkg#5865]
Issue due to geos linkage.

Portfile was (unconditionally) changed
```
file(TO_NATIVE_PATH "${CURRENT_INSTALLED_DIR}/lib/geos_c.lib" GEOS_LIBRARY_REL)
file(TO_NATIVE_PATH "${CURRENT_INSTALLED_DIR}/debug/lib/geos_cd.lib" GEOS_LIBRARY_DBG)
```
to
```
file(TO_NATIVE_PATH "${CURRENT_INSTALLED_DIR}/lib/libgeos.lib" GEOS_LIBRARY_REL)
file(TO_NATIVE_PATH "${CURRENT_INSTALLED_DIR}/debug/lib/libgeosd.lib" GEOS_LIBRARY_DBG)
```
which obviously fails in shared builds. (Why did CI not find that one? Ah CONTROL has not been updated for that port in #5839.....)

@Rastaban: This is your issue to fix ;)

**Comment by Neumann-A** [comment]
after fixing the missing file link issue I get unresolved symbols issues. #5839 should probably be reverted and resubmitted.

**Comment by Neumann-A** [comment]
[nmake-build-x64-windows-release-out.log](https://github.com/Microsoft/vcpkg/files/3021989/nmake-build-x64-windows-release-out.log) after changing back to

```
file(TO_NATIVE_PATH "${CURRENT_INSTALLED_DIR}/lib/geos_c.lib" GEOS_LIBRARY_REL)
file(TO_NATIVE_PATH "${CURRENT_INSTALLED_DIR}/debug/lib/geos_cd.lib" GEOS_LIBRARY_DBG)
```

**Comment by JuttaAnders** [comment]
@Neumann-A sorry for breaking the build.
I hope I fixed that in [https://github.com/Microsoft/vcpkg/pull/5882](https://github.com/Microsoft/vcpkg/pull/5882)

**Comment by wangli28** [comment]
Thanks for your feedback. This issue has been fixed in #5882. So I am closing this issue for now.
